xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/web/SaveAction.java

```diff
@@ -221,6 +241,80 @@ public boolean save(XWikiContext context) throws XWikiException
         return false;
     }
 
+    private DocumentRevisionProvider getDocumentRevisionProvider()
+    {
+        if (this.documentRevisionProvider == null) {
+            this.documentRevisionProvider = Utils.getComponent(DocumentRevisionProvider.class);
+        }
+
+        return this.documentRevisionProvider;
+    }
+
+    /**
+     * Check if the version of the document being saved is conflicting with another version.
+     * This check is done by getting the "previousVersion" parameter from the request and comparing it with
+     * latest version of the document. If the current version of the document is not the same as the previous one,
+     * a diff is computed on the document content: a conflict is detected only if the contents are different.
+     * @param context the current context of the request.
+     * @return true in case of conflict. If it's true, the answer is immediately sent to the client.
+     */
+    private boolean isConflictingWithVersion(XWikiContext context, XWikiDocument newDoc) throws XWikiException
+    {
+        XWikiRequest request = context.getRequest();
+        XWikiDocument doc = context.getDoc();
+
+        // the document is new we don't have to check the version date or anything
+        if ("true".equals(request.getParameter("isNew")) && doc.isNew()) {
```

Comment:
Be careful the context document is usually the default locale and not the translation you might be editing currently.

Reply:
Hmm right, I should use the document I have in argument.